### DISCUSSION TITLE
xgboost?

### DISCUSSION THREAD

**Discussion by zhihuba** [nautechsystems/nautilus_trader#1851] [General]
The description mentions machine learning. How can this be done using xgboost?

**Comment by cjdsellers** [comment]
Hi @zhihuba 

Thanks for your interest.

It's certainly possible. Nautilus allows you to write any arbitrary Python in your strategy (which isn't async) and will work the same way backtest and live.

So this would include many Python ML frameworks. Specific methods are out of scope in the docs though because there are so many options, and Nautilus remains unopinionated about this.

I hope that helps!

**Reply by zhihuba** [reply]
😂😂What is said. Nothing seems to be said. Add documentation if possible. Or examples
